src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,41 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_SIZE = 10;
+    private Object[] array = new Object[ARRAY_SIZE * 2];
+    private int iterator = 0;
+    private int size = 0;
```

Comment:
remove wasteful `iterator `field
```suggestion
    private int size;
```